#**Tutorial # 2 : Energy Balance**


**Week 1, Day 5, Climate Modeling**

**Content creators:** Jenna Pearson, Brodie Pearson, and Abigail Bodner

**Content reviewers:** Draco Xu

**Content editors:** TBD

**Production editors:** TBD

**Our 2023 Sponsors:** TBD

#**Tutorial Objectives**

In this tutorial students will learn about the components that define energy balance, including insolation and albedo.

By the end of this tutorial students will be able to:
* Calculate the albedo of Earth based on observations.
* Define and find the equilibrium temperature under the assumption of energy balance.

# Setup

## Package Imports

In [1]:
import xarray as xr                     # used to manipulate data and open datasets
import numpy as np                      # used for algebra and array operations
import matplotlib.pyplot as plt         # used for plotting

## Figure Settings

In [2]:
import ipywidgets as widgets       # interactive display
%config InlineBackend.figure_format = 'retina'
plt.style.use("https://raw.githubusercontent.com/ClimateMatchAcademy/course-content/main/cma.mplstyle")

# Section 1 : Insolation and Albedo

Just as Earth emits radiation, so does the sun. The incoming solar radiation, called **insolation**, is observed to be $Q = 340 W m^{-2}$. 

Some of this radiation is reflected back to space (for example off of ice and snow). From observations the amount reflected back is $F_{ref} = 100 W m^{-2}$. The *fraction* of reflected radiation is captured by the **albedo (**$\mathbf{\alpha}$**)**

\begin{align}
\alpha = \frac{F_{ref}}{Q}
\end{align}

Albedo is a unitless number between 0 and 1. We can use this formula and python to find the albedo of Earth.

In [3]:
# define the observed insolation based on observations from the IPCC AR6 Figure 7.2
Q = 340 # W m^-2

# define the observed reflected radiation based on observations from the IPCC AR6 Figure 7.2
F_ref = 100 # W m^-2

# plug into equation
alpha = (F_ref/Q) # unitless number between 0 and 1

# display answer
print('Albedo: ',alpha)

Albedo:  0.29411764705882354


## Questions

1. What does a low albedo imply? What about a high albedo?

## Coding Exercises

1. By using a for loop, calculate and print the $\alpha$ for three values of $OLR = 300,340,380$.

In [15]:
#################################################
## TODO for students: details of what they should do ##
# Fill out function and remove
raise NotImplementedError("Student exercise: By using a for loop, calculate and pring the albedo for three values of OLR = 300,340,380.")
#################################################
  

# define the observed reflected radiation based on observations from the IPCC AR6 Figure 7.2
F_ref = ...

# loop through values of insolation
for Q in [...]:

  # plug into equation
  alpha = ...

  # display answer
  print('Insolation:',Q,',        ','albedo: ' ,alpha)

NotImplementedError: ignored

In [13]:
# to_remove solution

# define the observed reflected radiation based on observations from the IPCC AR6 Figure 7.2
F_ref = 100 # W m^-2

# loop through values of insolation
for Q in [300,340,380]:

  # plug into equation
  alpha = (F_ref/Q) # unitless number between 0 and 1

  # display answer
  print('Insolation:',Q,',        ','albedo: ' ,alpha)

Insolation: 300 ,         albedo:  0.3333333333333333
Insolation: 340 ,         albedo:  0.29411764705882354
Insolation: 380 ,         albedo:  0.2631578947368421


# Section 2 : Absorbed Shortwave Radiation

The **absorbed shortwave radiation (ASR)** is the amount of this isolation that is *not* reflected, and actually makes it to Earth's surface. Thus,

\begin{align}
ASR = Q-F_{ref} = (1-\alpha)Q
\end{align}

Using observations and python we can esimate the absorbed shortwave radiation.

In [16]:
# define the observed insolation based on observations from the IPCC AR6 Figure 7.2
Q = 340 # W m^-2

# define the observed reflected radiation based on observations from the IPCC AR6 Figure 7.2
F_ref = 100 # W m^-2

# plug into equation
alpha = (F_ref/Q) # unitless number between 0 and 1

# plug into equation
ASR = (1-alpha)*Q

# display answer
print('Absorbed Shortwave Radiation: ',ASR,' W m^-2')

Absorbed Shortwave Radiation:  239.99999999999997  W m^-2


### Questions

1.  Compare the value of ASR to the observed OLR of $239 W m^{-2}$. Is it more or less?
2. Does this model take into account any effects of gases in that atmosphere? Are there any you might think are impotant that should be included in more complex models?

# Section 3 : Equilibrium Temperature

Energy Balance is achieved when radiation absorbed by Earth's surface (ASR) is equal to longwave radiation going out to space (OLR). That is 

\begin{align}
ASR = OLR
\end{align}

By substituting in the equations from previous sections, we can find the surface temperature of Earth needed to maintain this balance. This is called the **equilibrium temperature (** $\mathbf{T_{eq}}$ **)**.

Recall $OLR = \tau\sigma T^4$ and $ASR = (1-\alpha)Q$. The **equilibrium temperature** is the temperature the system would have if energy balance was perfectly reached. Assuming energy balance, we will call the emission temperature denoted previously the equilibrium temperature ($T_{eq}$) instead. Thus,

\begin{align}
(1-\alpha)Q = ASR = OLR = \tau\sigma T_{eq}^4
\end{align}

Solving for $T_{eq}$ we find

\begin{align}
T_{eq} = \sqrt[4]{\frac{(1-\alpha)Q}{\tau\sigma}}
\end{align}

We can implement this equation in python to find the numeric value.

In [17]:
# define the observed insolation based on observations from the IPCC AR6 Figure 7.2
Q = 340 # W m^-2

# define the observed reflected radiation based on observations from the IPCC AR6 Figure 7.2
F_ref = 100 # W m^-2

# define albedo
alpha = (F_ref/Q) # unitless number between 0 and 1

# define the Stefan-Boltzmann Constant, noting we are using 'e' for scientific notation
sigma = 5.67e-8 # W m^-2 K^-4

# define transmissivity (calculated previously from observations in tutorial 1)
tau = 0.6127 # unitless number between 0 and 1

# plug into equation
T_eq = (((1-alpha)*Q)/(tau * sigma))**(1/4)

# display answer
print('Equilibrium Temperature: ',T_eq,'K or',T_eq - 273, 'C')

Equilibrium Temperature:  288.300287595812 K or 15.300287595811994 C


## Questions

1.  Is the calculated equilibrium temperature warmer or cooler than the observed surface temperature of $288 K$?
2.  What do you think is responsible for the difference? Refer to previous sections if needed.

# Section 4 : Climate Change Scenario

Assume due to the increasing presence of greenhouse gases in the the atmosphere, that $\tau$ decreases to $0.57$. 

We can then use our climate model and python to find the new equilibrium temperature.

In [18]:
# define the observed insolation based on observations from the IPCC AR6 Figure 7.2
Q = 340 # W m^-2

# define the observed reflected radiation based on observations from the IPCC AR6 Figure 7.2
F_ref = 100 # W m^-2

# define albedo
alpha = (F_ref/Q) # unitless number between 0 and 1

# define the Stefan-Boltzmann Constant, noting we are using 'e' for scientific notation
sigma = 5.67e-8 # W m^-2 K^-4

# define transmissivity (assupmtion in this case)
tau = 0.57 # unitless number between 0 and 1

# plug into equation
T_eq = (((1-alpha)*Q)/(tau * sigma))**(1/4)

# display answer
print('New Equilibrium Temperature: ',T_eq,'K or',T_eq - 273, 'C')

New Equilibrium Temperature:  293.5542225759401 K or 20.554222575940116 C


## Questions
1.  Does a reduction in the transmissivity, $\tau$, imply a decrease or increase in OLR?
2.  How does the new equilibrium temperature compare to that calculated previously?


## Coding Exercises

1.  Using a for loop, calculate the equilibrium temperatures for $\tau = 0.2,0.4,0.8$.

In [20]:
#################################################
## TODO for students: details of what they should do ##
# Fill out function and remove
raise NotImplementedError("Student exercise: By using a for loop, calculate the equilibrium temperature for tau = 0.2, 0.4, and 0.8.")
#################################################
  
  # define the observed insolation based on observations from the IPCC AR6 Figure 7.2
Q = 340 # W m^-2

# define the observed reflected radiation based on observations from the IPCC AR6 Figure 7.2
F_ref = 100 # W m^-2

# define albedo
alpha = (F_ref/Q) # unitless number between 0 and 1

# define the Stefan-Boltzmann Constant, noting we are using 'e' for scientific notation
sigma = 5.67e-8 # W m^-2 K^-4

# loop through values of tau
for tau in [...]:

  # plug into equation
  T_eq = ...

  # display answer
  print('Transmissivity:',tau,',     ' 'Equilibrium Temperature: ' ,T_eq, 'K')

NotImplementedError: ignored

In [21]:
# define the observed insolation based on observations from the IPCC AR6 Figure 7.2
Q = 340 # W m^-2

# define the observed reflected radiation based on observations from the IPCC AR6 Figure 7.2
F_ref = 100 # W m^-2

# define albedo
alpha = (F_ref/Q) # unitless number between 0 and 1

# define the Stefan-Boltzmann Constant, noting we are using 'e' for scientific notation
sigma = 5.67e-8 # W m^-2 K^-4

# loop through values of tau
for tau in [0.2,0.4,0.8]:

  # plug into equation
  T_eq = (((1-alpha)*Q)/(tau * sigma))**(1/4)

  # display answer
  print('Transmissivity:',tau,',     ' 'Equilibrium Temperature: ' ,T_eq, 'K')

Transmissivity: 0.2 ,     Equilibrium Temperature:  381.4165616365676 K
Transmissivity: 0.4 ,     Equilibrium Temperature:  320.73181939858716 K
Transmissivity: 0.8 ,     Equilibrium Temperature:  269.70223719007373 K
